### PR TITLE
Add file download functionalities

### PR DESCRIPTION
**Pull request by lawiet47**
Adds file retriaval functionalies as described here: https://docs-cortex.paloaltonetworks.com/r/Cortex-XDR/Cortex-XDR-API-Reference/File-Retrieval-Details

### CODE REVIEW

File: cortex_xdr_client/api/base_api.py [hunk 1]
```diff
@@ -13,11 +13,11 @@ def __init__(self, auth: Authentication, fqdn: str, api_name: str, timeout: Tupl
         self._requests_timeout = timeout
         self._api_name = api_name
 
-    def _get_url(self, call_name: str) -> str:
+    def _get_url(self, call_name: Optional[str]) -> str:
```
Comment (ebarti) [inline cortex_xdr_client/api/base_api.py:16]: Why is this now optional?

Comment (lawiet47) [reply cortex_xdr_client/api/base_api.py]: Forgot to change it back once I implemented a different solution for download_api.

File: cortex_xdr_client/api/base_api.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def _get_url(self, call_name: str) -> str:
         return f"https://api-{self._fqdn}/public_api/v1/{self._api_name}/{call_name}"
 
     def _call(self,
-              call_name: str,
+              call_name: Optional[str] = None,
```
Comment (ebarti) [inline cortex_xdr_client/api/base_api.py:20]:
```suggestion
              call_name:str = None,
```

File: cortex_xdr_client/api/base_api.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ def _get_url(self, call_name: str) -> str:
         return f"https://api-{self._fqdn}/public_api/v1/{self._api_name}/{call_name}"
 
     def _call(self,
-              call_name: str,
+              call_name:str,
```
Comment (ebarti) [inline cortex_xdr_client/api/base_api.py:20]:
```suggestion
              call_name: str,
```

**Review by ebarti — APPROVED** [review]

**Pull request merged**